### PR TITLE
add stressng vm support and update main and stressng readme file

### PR DESCRIPTION
**Pull request by ebattat**
Adding CNV support for stressng, test changes against Alias lab using OCP4.6/CNV2.5.3

### CODE REVIEW

File: docs/stressng.md [hunk 1]
```diff
@@ -65,3 +65,55 @@ Once done creating/editing the resource file, you can run it by:
 # kubectl apply -f <path_to_cr_file>
 ```
 
+## Running stressng in VMs through kubevirt/cnv [Preview]
+Note: this is currently in preview mode.
+you can modify [cr.yaml](../resources/crds/ripsaw_v1alpha1_stressng_vm.yaml) to your needs.
+
+
+### Pre-requisites
```
Comment (jtaleric) [inline docs/stressng.md:73]: Optional*

File: docs/stressng.md [hunk 2]
```diff
@@ -65,3 +65,55 @@ Once done creating/editing the resource file, you can run it by:
...
+
+You must have configured your k8s cluster with [Kubevirt](https://kubevirt.io) preferably v0.23.0 (last tested version).
```
Comment (jtaleric) [inline docs/stressng.md:75]: if you want to run with VMs, however it isn't totally necessary.

File: roles/stressng/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+resource_kind: "{{ workload.args.kind | default('pod') }}"
+uperf:
```
Comment (jtaleric) [inline roles/stressng/defaults/main.yml:3]: Bad copy and paste?

**Review by jtaleric — CHANGES_REQUESTED** [review]
Did you also test w/o VMs?

File: docs/stressng.md (hunk 1 above)
Comment (ebattat) [reply docs/stressng.md:73]: Yes, test both ripsaw_v1alpha1_stressng_cr.yaml and ripsaw_v1alpha1_stressng_vm.yaml

File: resources/crds/ripsaw_v1alpha1_stressng_cr.yaml [hunk 4]
```diff
@@ -1,15 +1,15 @@
 apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
 kind: Benchmark
 metadata:
-  name: stressng
+  name: stressng-benchmark
   namespace: my-ripsaw
 spec:
   elasticsearch:
-    url: http://my.elasticsearch.server:80
+    url: ""
```
Comment (rsevilla87) [inline resources/crds/ripsaw_v1alpha1_stressng_cr.yaml:8]: All examples are standarized to this URL; we don't should change it

File: roles/stressng/tasks/main.yaml [hunk 5]
```diff
@@ -66,22 +77,38 @@
...
+        status:
+          state: Benchmark Complete
+          complete: true
+        when: stressng_workload_pod | json_query('resources[].status.succeeded')"
+    when: resource_kind == "pod"
 
+  - block:
+      - name: get complete
+        command: "redis-cli get complete"
+        register: complete_status
```
Comment (rsevilla87) [inline roles/stressng/tasks/main.yaml:103]: Add a new line "\n"

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
Added a couple of small nits, also needs rebase @ebattat

File: resources/crds/ripsaw_v1alpha1_stressng_cr.yaml (hunk 4 above)
Comment (ebattat) [reply resources/crds/ripsaw_v1alpha1_stressng_cr.yaml]: Fix it in the following commit:
https://github.com/cloud-bulldozer/benchmark-operator/pull/522/commits/4d5c0fbe895773f08fd8594dc49696d15f8e8eff

File: roles/stressng/tasks/main.yaml (hunk 5 above)
Comment (ebattat) [reply roles/stressng/tasks/main.yaml:103]: Fix it in the following commit:
https://github.com/cloud-bulldozer/benchmark-operator/pull/522/commits/c2b61762ecca7eb5b47ed598536bde610420a124

**Comment by ebattat** [comment]
Hi,

I have rebase and resolved conflicts in the last commit:
https://github.com/cloud-bulldozer/benchmark-operator/pull/522/commits/cf27706f1fc99014d386bb3a9f16ce089d04e83d

Please let me know if that ok.

Best regards,
Eli Battat.

**Review by rsevilla87 — APPROVED** [review]
LGTM!

PTAL @jtaleric

**Comment by ebattat** [comment]
Hi,

Yes, tested w/o VMs

Best regards,
Eli Battat.

**Pull request merged**